Client/src/components/navbar.tsx

```diff
@@ -0,0 +1,186 @@
+    setModalOpen(!modalOpen);
+  };
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    (async function evalAuth() {
+      if (isAuthenticated === true) {
+        const { user, message } = await apiService.profile();
+        console.log(user)
+        if (user) dispatch(refreshData(user));
+        else console.log(message);
+      }
+    })();
+  }, [isAuthenticated]);
+
+  return (
+    <>
+      <div className="navbar">
+        <div className="navbar-start">
+          <button className="button" onClick={() => navigate('/')}>
+            aMAZEthing
+          </button>
+          <MediaQuery minWidth={951}>
+            {location.pathname === '/game' && (
+              <button className="button" onClick={() => navigate('/learning')}>
+                LEARN
+              </button>
+            )}
+            {location.pathname === '/learning' && (
+              <button className="button" onClick={() => navigate('/game')}>
+                PLAY
+              </button>
+            )}
+            {location.pathname === '/profile' && (
+              <button className="button" onClick={() => navigate('/game')}>
+                PLAY
+              </button>
+            )}
+            {location.pathname === '/profile' && (
+              <button className="button" onClick={() => navigate('/learning')}>
+                LEARN
+              </button>
+            )}
+          </MediaQuery>
+        </div>
+        <div className="navbar-end">
+          <MediaQuery minWidth={951}>
+            {location.pathname === '/' && (
+              <button className="button" onClick={() => navigate('/about')}>
+                ABOUT
+              </button>
+            )}
+            {isAuthenticated && !(location.pathname === '/profile') && (
+              <button className="button" onClick={() => navigate('/profile')}>
+                PROFILE
+              </button>
+            )}
+            {!isLoading && !user && (
+              <button className="button" onClick={() => loginWithRedirect()}>
+                LOGIN
+              </button>
+            )}
+            {!isLoading && user && (
+              <button className="button" onClick={() => logout()}>
+                LOGOUT
+              </button>
+            )}
+          </MediaQuery>
+          <MediaQuery maxWidth={950}>
+            <div
+              className="container"
+              onClick={() => {
+                toggleNavbar();
+                toggleModal();
+              }}
+            >
+              <div className={`bar1 ${toggle && 'active'}`} />
+              <div className={`bar2 ${toggle && 'active'}`} />
+              <div className={`bar3 ${toggle && 'active'}`} />
+            </div>
+          </MediaQuery>
+        </div>
+        <MediaQuery maxWidth={950}>
+          <div className={`modal-body open-${modalOpen}`}>
+            {location.pathname === '/' && (
+              <button
+                className="button"
```

Comment:
We can create a function with a switch that does the toogleNavbar and Model at the end. and goes to the location depending on the argument passed - NEW TASK

Reply:
Done